src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7859,8 +7859,22 @@ private struct REPARSE_GUID_DATA_BUFFER
             FileAttributes dwFlagsAndAttributes,
             IntPtr hTemplateFile);
 
+        internal sealed class SafeFindHandle : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private SafeFindHandle() : base(true) { }
+
+            protected override bool ReleaseHandle()
+            {
+                return FindClose(this.handle);
+            }
+
+            [DllImport(PinvokeDllNames.FindCloseDllName)]
+            [return: MarshalAs(UnmanagedType.Bool)]
+            private static extern bool FindClose(IntPtr handle);
+        }
+
         [DllImport(PinvokeDllNames.FindFirstFileDllName, EntryPoint = "FindFirstFileExW", SetLastError = true, CharSet = CharSet.Unicode)]
-        private static extern SafeFileHandle FindFirstFileEx(string lpFileName, FINDEX_INFO_LEVELS fInfoLevelId, ref WIN32_FIND_DATA lpFindFileData, FINDEX_SEARCH_OPS fSearchOp, IntPtr lpSearchFilter, int dwAdditionalFlags);
+        private static extern SafeFindHandle FindFirstFileEx(string lpFileName, FINDEX_INFO_LEVELS fInfoLevelId, ref WIN32_FIND_DATA lpFindFileData, FINDEX_SEARCH_OPS fSearchOp, IntPtr lpSearchFilter, int dwAdditionalFlags);
```

Comment:
I would add text from PR description as a comment here. Otherwhise looks OK,.

Reply:
I think the return type makes it self documenting

Reply:
> SetLastError = true

It seems we don't use this.

Reply:
I'd keep the SetLastError anyway, to reduce the risk of bugs if the declaration is later copied elsewhere. Or does PowerShell generally have a policy of removing unused DllImportAttribute.SetLastError for perf reasons?

Reply:
> for perf reasons

It was my thought but I don't measure.